packages/block-library/src/navigation/index.php

```diff
@@ -67,6 +67,521 @@ function block_core_navigation_sort_menu_items_by_parent_id( $menu_items ) {
+			$parsed_blocks           = block_core_navigation_parse_blocks_from_menu_items( $menu_items_by_parent_id[0], $menu_items_by_parent_id );
+			$inner_blocks            = new WP_Block_List( $parsed_blocks, $attributes );
+		}
+
+		// Load inner blocks from the navigation post.
+		if ( array_key_exists( 'ref', $attributes ) ) {
+			$navigation_post = get_post( $attributes['ref'] );
+			if ( ! isset( $navigation_post ) ) {
+				return '';
+			}
+
+			// Only published posts are valid. If this is changed then a corresponding change
+			// must also be implemented in `use-navigation-menu.js`.
+			if ( 'publish' === $navigation_post->post_status ) {
+				$parsed_blocks = parse_blocks( $navigation_post->post_content );
+
+				// 'parse_blocks' includes a null block with '\n\n' as the content when
+				// it encounters whitespace. This code strips it.
+				$compacted_blocks = block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
+
+				// TODO - this uses the full navigation block attributes for the
+				// context which could be refined.
+				$inner_blocks = new WP_Block_List( $compacted_blocks, $attributes );
+			}
+		}
+
+		// If there are no inner blocks then fallback to rendering an appropriate fallback.
+		if ( empty( $inner_blocks ) ) {
+			$fallback_blocks = block_core_navigation_get_fallback_blocks();
+
+			// Fallback my have been filtered so do basic test for validity.
+			if ( empty( $fallback_blocks ) || ! is_array( $fallback_blocks ) ) {
+				return '';
+			}
+
+			$inner_blocks = new WP_Block_List( $fallback_blocks, $attributes );
+		}
+
+		/**
+		 * Filter navigation block $inner_blocks.
+		 * Allows modification of a navigation block menu items.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param \WP_Block_List $inner_blocks
+		 */
+		$inner_blocks = apply_filters( 'block_core_navigation_render_inner_blocks', $inner_blocks );
+
+		$post_ids = block_core_navigation_get_post_ids( $inner_blocks );
+		if ( $post_ids ) {
+			_prime_post_caches( $post_ids, false, false );
+		}
+
+		return $inner_blocks;
+	}
+
+	/**
+	 * Gets the name of the current navigation, if it has one.
+	 */
+	private static function get_navigation_name( $attributes, &$seen_menu_names ) {
+		$navigation_name = $attributes['ariaLabel'] ?? '';
+
+		// Load the navigation post.
+		if ( array_key_exists( 'ref', $attributes ) ) {
+			$navigation_post = get_post( $attributes['ref'] );
+			if ( ! isset( $navigation_post ) ) {
+				return $navigation_name;
+			}
+
+			// Only published posts are valid. If this is changed then a corresponding change
+			// must also be implemented in `use-navigation-menu.js`.
+			if ( 'publish' === $navigation_post->post_status ) {
+				$navigation_name = $navigation_post->post_title;
+
+				if ( isset( $seen_menu_names[ $navigation_name ] ) ) {
+					++$seen_menu_names[ $navigation_name ];
+				} else {
+					$seen_menu_names[ $navigation_name ] = 1;
+				}
```

Comment:
We need a comment around this. Or even better a method which handles this code.

Reply:
Done in 08ec8ddb08